lib/decoder.js

```diff
@@ -319,6 +319,18 @@ var JpegImage = (function jpegImage() {
         }
       }
 
+      if (mcu === mcuExpected) {
+        // Skip trailing bytes at the end of the scan - until we reach the next marker
+        do {
+          if (data[offset] === 0xFF) {
+            if (data[offset + 1] !== 0x00) {
```

Comment:
mind walking me through this one? :) 

if we're intending on being loose AFAICT we'd be fine just checking for `data[offset] === 0xFF`?

Reply:
According to the spec, any `FF` followed by `00` is not a marker. Since I don’t have enough examples to know what the heck encoders might use as fill bytes, I want to be sure that I’m not stopping before finding an actual marker.

Reply:
But the code could be more concise. It’s a leftover from having more conditions there before arriving at this version.

Reply:
it's definitely not a marker, but I was under the impression it was filler data. it is handled appropriately in the rest of jpeg-js fwiw

but we basically cycle through it just like we would here, so it wouldn't really have a practical effect other than consistency with the code below

Reply:
maybe we make a note of this conversation in a comment or change it so we don't wonder in the future why FF00 is a valid marker on line 336 but not here? :)